pkg/process/metadata/workloadmeta/extractor.go

```diff
@@ -74,11 +85,33 @@ func (w *WorkloadMetaExtractor) Extract(procs map[int32]*procutil.Process) {
 	}
 
 	deadProcs := getDifference(w.cache, newCache)
-	w.grpcListener.writeEvents(deadProcs, newEntities)
 
 	w.cacheMutex.Lock()
 	w.cache = newCache
+	w.cacheVersion++
 	w.cacheMutex.Unlock()
+
+	// Drop previous cache diff if it hasn't been consumed yet
+	select {
+	case <-w.diffChan:
+		// drop message
+		break
+	default:
```

Comment:
Perhaps a warning log or some telemetry could be useful to catch this issue

Reply:
I suppose it would also be possible to simply reset the grpc connection when it occurs

Reply:
An old diff can also be dropped if there's no consumer for it yet. For example, if the core agent takes longer to start than the process agent.

I've added a debug message to catch when this happen for now and we can add a telemetry for it later.